src/storage-client/src/controller/metrics_scraper.rs

```diff
@@ -0,0 +1,136 @@
+// Copyright Materialize, Inc. and contributors. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+//! A tokio task (and support machinery) for maintaining storage-managed
+//! collections.
+
+use futures::stream::BoxStream;
+use futures::StreamExt;
+
+use itertools::Itertools;
+use mz_orchestrator::ServiceProcessMetrics;
+use mz_ore::cast::CastFrom;
+use mz_repr::{Datum, GlobalId, Row};
+use timely::progress::ChangeBatch;
+use tokio::sync::oneshot;
+
+use crate::controller::collection_mgmt::CollectionManager;
+
+#[derive(Debug)]
+pub struct MetricsScraper {
+    shutdown_button: Option<oneshot::Sender<()>>,
+}
```

Comment:
this is a bit of a weird api, as its just a drop token! I think you should just make `MetricsScraper::new` into a function that returns a `Box<dyn Any + Send + Sync`, placing the sender in that box. Dropping that will signal to the receiver to finish (it will return `Err(ChannelClosed)`, without needing to do any of the `send`-ing you do in the `Drop` impl below!

Reply:
Will do! I'll also rename `new` to `start`